pkg/models/node_approval.go

```diff
@@ -0,0 +1,39 @@
+package models
+
+import "fmt"
+
+// NodeApproval is used to denote the approval status of a given
+// node. These values are set based on the approval process for
+// nodes which will auto-approve some, rely on human approval for
+// others as well as rejecting some nodes as inelligible.
+type NodeApproval string
+
+const (
+	NodeApprovalUnknown  NodeApproval = "unknown"
+	NodeApprovalPending  NodeApproval = "pending"
+	NodeApprovalApproved NodeApproval = "approved"
+	NodeApprovalRejected NodeApproval = "rejected"
+)
+
+// ParseApproval accepts a string representation of a NodeApproval
+// and returns the matching NodeApproval value, or an error if
+// the string cannot be matched to a NodeApproval value.
+func ParseApproval(s string) (NodeApproval, error) {
+	switch s {
+	case "pending":
+		return NodeApprovalPending, nil
+	case "approved":
+		return NodeApprovalApproved, nil
+	case "rejected":
+		return NodeApprovalRejected, nil
+	case "unknown":
+		return NodeApprovalUnknown, nil
+	default:
```

Comment:
what are the possible cases to have "unknown" state?

Reply:
Originally it was going to be the return value for the Parse() method, but then I thought it would be a reasonable state for nodes until I add the approval endpoints.